frontend/src/routes/Profile.jsx

```diff
@@ -0,0 +1,437 @@
+      return false;
+    }
+    if (!formData.LastName) {
+      setValidationMessage('Last name is required');
+      return false;
+    }
+    if (!formData.Username) {
+      setValidationMessage('Username is required');
+      return false;
+    }
+    if (!emailRegex.test(formData.Email)) {
+      setValidationMessage('Email is invalid!');
+      return false;
+    }
+    setValidationMessage('');
+    return true;
+  }
+
+  const handleSaveClick = () => {
+    if (!isValid()) {
+      return;
+    }
+    hasChanged(true);
+    onSave(formData);
+  };
+
+  return (
+    <div className="modal">
+      <div className="modalContent">
+        <h1 className="editProfileText">Edit Profile</h1>
+        <p>First Name:</p>
+        <input
+          className="inputEdit"
+          name="FirstName"
+          value={formData.FirstName}
+          onChange={handleChange}
+        />
+        <p>Last Name:</p>
+        <input
+          className="inputEdit"
+          name="LastName"
+          value={formData.LastName}
+          onChange={handleChange}
+        />
+        <p>Username:</p>
+        <input
+          className="inputEdit"
+          name="Username"
+          value={formData.Username}
+          onChange={handleChange}
+        />
+        <p>Email:</p>
+        <input
+          className="inputEdit"
+          name="Email"
+          value={formData.Email}
+          onChange={handleChange}
+        />
+        <p className="errorMessage">{validationMessage}</p>
+        <button className="EditWindowButton" onClick={handleSaveClick}>
+          Accept
+        </button>
+        <button className="EditWindowButton" onClick={onClose}>
+          Cancel
+        </button>
+      </div>
+    </div>
+  );
+}
+EditWindow.propTypes = {
+  userInfo: PropTypes.shape({
+    FirstName: PropTypes.string,
+    LastName: PropTypes.string,
+    Username: PropTypes.string,
+    Email: PropTypes.string,
+    Bio: PropTypes.string
+  }).isRequired,
+  onSave: PropTypes.func.isRequired,
+  onClose: PropTypes.func.isRequired,
+  hasChanged: PropTypes.func.isRequired
+};
+
+function PasswordChange({
+  onSave,
+  onClose,
+  hash,
+  setNewPasswordHash,
+  hasChanged
+}) {
+  const [validationMessage, setValidationMessage] = useState('');
+  const [formData, setFormData] = useState({
+    currentPassword: '',
+    newPassword: '',
+    confirmNewPassword: ''
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({ ...prevData, [name]: value }));
+  };
+
+  async function hashPassword() {
+    let hashed = await getHash(formData.currentPassword);
+    return hashed;
+  }
+
+  function isValid(oldHash, newHash) {
+    if (formData.newPassword !== formData.confirmNewPassword) {
+      setValidationMessage('Passwords do not match');
+      return false;
+    }
+
+    //Until fetch works tempPass will bypass validation, needs to be removed later on!
```

Comment:
whitespace

Reply:
also add a `// TODO: Until fetch works...`. Most IDEs/editors will automatically highlight TODOs in code (and if not, by searching for TODO you can find all the todos easily :))